extensions/amp-story-captions/validator-amp-story-captions.protoascii

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+tags: {  # amp-story-captions
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-story-captions"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-story-captions>
```

Comment:
Any additional constraints? e.g.
1. How many `<amp-story-captions>` may appear in an `<amp-story>`?
2. Needs to be first or last child?
3. Constraints on what children it can have?

Reply:
1. Any number, but if there's no amp-video pointing to it, it's useless (so effectively <= the number of amp-video elements). Maybe we could require an id attribute and if possible, require that an amp-video uses this id.
2. It can be anywhere (directly inside of a amp-story-grid-layer, nested in a div, etc.).
3. It can't have any children. The only content is the captions that it renders.

Reply:
@twifkak ping for reply to answers

Reply:
Oof, I'm so sorry for the delay. Thanks for pinging me @honeybadgerdontcare.

1. Ah OK. Yeah, requiring "id" sounds like an easy win, if it is indeed required. I don't think we have a way of requiring that the id is used by an amp-video, but I think we can require the reverse: if an amp-video uses a story-captions id, then the corresponding story-captions exists. That's the [ValueSet fields](https://github.com/ampproject/amphtml/blob/aa1f758a1c81213bec75d51654e69d6e4d64edd8/validator/validator.proto#L145-L161). Just an optional idea. Probably better as a follow-up PR since it requires changes to the validator.proto file, and I forget whether that requires changes be staged in a certain order...
2. :+1: 
3. Maybe set [mandatory_num_child_tags](https://github.com/ampproject/amphtml/blob/aa1f758a1c81213bec75d51654e69d6e4d64edd8/validator/validator.proto#L418) to 0, then? Obviously double-check that the tests are still happy after that. I won't block on that since it's not critical & I feel bad for the delay.

Reply:
For the id attribute rule, do I need a long disallowed_value_regex like this? https://github.com/ampproject/amphtml/blob/main/extensions/amp-script/validator-amp-script.protoascii#L39

Reply:
Yes. See https://github.com/ampproject/amphtml/blob/main/validator/validator-main.protoascii#L3940

Reply:
But you could use the attr_list and not duplicate it again, unless there's a special need to have it more customized.